internal/photoprism/mediafile.go

```diff
@@ -378,6 +378,28 @@ func (m *MediaFile) ExtractEmbeddedVideo() (string, error) {
 
 			return dstPath, nil
 		}
+	} else if metaData.Error == nil && metaData.EmbeddedVideoOffset > 0 {
```

Comment:
The if case above, `metaData.EmbeddedVideo != ""` was something I added for Samsung motion photos. Which means we can utilize it here as well, we don't need a separate else if case with `metaData.EmbeddedVideoOffset`. If you look at my most recent [changes](https://github.com/photoprism/photoprism/pull/3709/files#diff-e5a39f92547299cd358543ff78e747595e351aa0f56a50d43c0034cb94465d8aL334), I changed this function a bit to accommodate both Samsung and Google photos as they can be extracted in the same way, we just need to decide if we will 1) use the directory item length approach or 2) use the approach for finding the MP4 header and then we don't need to have this duplicate else if case.

Reply:
This PR is now updated to handle samsung/google the same, let me know if you still have a concern/suggestion for the way its done now